packages/builder/src/pages/builder/app/[application]/design/[screenId]/_components/AppPreview.svelte

```diff
@@ -130,7 +130,7 @@
       // Catch any app errors
       loading = false
       error = event.error || "An unknown error occurred"
-    } else if (type === "select-component" && data.id) {
+    } else if (type === "select-component" && "id" in data) {
```

Comment:
We can remove any changes to this file if we opt for sending up the screen ID as opposed to `null`.